interface-impl-amf/src/main/java/org/mule/amf/impl/model/ParameterValidationStrategyFactory.java

```diff
@@ -17,14 +18,16 @@ private ParameterValidationStrategyFactory() {
     throw new IllegalStateException("Utility class");
   }
 
-  static ParameterValidationStrategy getStrategy(AnyShape anyShape, boolean schemaNeedsQuotes) {
-    return isYamlValidationNeeded(anyShape) ? new YamlParameterValidationStrategy(anyShape, schemaNeedsQuotes)
-        : getJsonParameterValidationStrategy(anyShape, schemaNeedsQuotes);
+  static ParameterValidationStrategy getStrategy(AnyShape anyShape, boolean needsCharEscaping,
+                                                 AMFConfiguration amfConfiguration) {
+    return isYamlValidationNeeded(anyShape) ? new YamlParameterValidationStrategy(anyShape, needsCharEscaping, amfConfiguration)
```

Comment:
I would like for both the Yaml and JSON strategy to receive either the element client OR the configuration object. It looks weird for it to have different parameterization.